src/effects/lv2/lv2backend.cpp

```diff
@@ -0,0 +1,96 @@
+#include "effects/lv2/lv2backend.h"
+#include "effects/lv2/lv2manifest.h"
+
+LV2Backend::LV2Backend(QObject* pParent)
+        : EffectsBackend(pParent, tr("LV2")) {
+    m_pWorld = lilv_world_new();
+    initializeProperties();
+    lilv_world_load_all(m_pWorld);
+    enumeratePlugins();
+}
+
+LV2Backend::~LV2Backend() {
+    foreach(LilvNode* node, m_properties) {
+        lilv_node_free(node);
+    }
+    lilv_world_free(m_pWorld);
+    foreach(LV2Manifest* lv2Manifest, m_registeredEffects) {
+        delete lv2Manifest;
+    }
+}
+
+void LV2Backend::enumeratePlugins() {
+    const LilvPlugins *plugs = lilv_world_get_all_plugins(m_pWorld);
+    LILV_FOREACH(plugins, i, plugs) {
+        const LilvPlugin *plug = lilv_plugins_get(plugs, i);
+        if (lilv_plugin_is_replaced(plug)) {
+            continue;
+        }
+        LV2Manifest* lv2Manifest = new LV2Manifest(plug, m_properties);
```

Comment:
This should use lilv_plugin_has_latency to filter out plugins that add latency. We cannot compensate for latency in effects units with switchable input paths because we cannot suddenly switch which channels are being delayed during playback.

Reply:
lilv_plugin_has_latency just tells us if the Plug-In reports latency, not if it introduces it. 
There is nothing wrong to use a Plug-In that introduces Latency if the Mix knob sticks on fully wet. This is also the case for all our fidlib based effects, which introduce ad group delay.

Reply:
Mixxx can't do anything about plugins that add latency but do not report it. Plugins that add latency are problematic because activating them will create an audible gap in the signal. Even if that is done when the volume fader is down, it would create an odd situation to manipulate different signal paths with different latencies and throw off the alignment of waveforms.

Reply:
It depend on the amount of latency it will add. Currently we fade from the wet to the dry signal by one buffer size. This should cover all known delays. Do you know a plug-In that has a longer latency?